### PR TITLE
nft implementation

### PR DESCRIPTION
**Pull request by willeslau**
NFT creation with VEc<u8> as data input for object place holder.

### CODE REVIEW

File: filecoindot-nft/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,147 @@
+// Copyright 2021 ChainSafe Systems
+// SPDX-License-Identifier: LGPL-3.0-only
+
+//! # Filecoin Governance Pallet
```
Comment (ivanceras) [inline filecoindot-nft/src/lib.rs:4]: This should have a different doc-comment regarding NFT

File: filecoindot-nft/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        #[pallet::weight(T::WeightInfo::mint())]
+        pub fn mint(origin: OriginFor<T>, cid: Vec<u8>, proof: Vec<Vec<u8>>) -> DispatchResult {
+            let who = ensure_signed(origin)?;
+            orml_nft::Pallet::<T>::mint(
+                &who,
+                T::DefaultClassId::get(),
+                vec![],
+                TokenData::new(cid, proof),
+            )?;
+            Self::deposit_event(Event::MintedToken(who, T::DefaultClassId::get(), 1));
```
Comment (ivanceras) [inline filecoindot-nft/src/lib.rs:102]: Why is the `TokenId` 1 here? If it is a const, it should be declared as const above.

File: filecoindot-nft/src/lib.rs (hunk 1 above)
Comment (willeslau) [reply filecoindot-nft/src/lib.rs:4]: oops, you got me

File: filecoindot-nft/src/lib.rs (hunk 2 above)
Comment (willeslau) [reply filecoindot-nft/src/lib.rs:102]: nice nice, will convert to const

**Review by ivanceras — CHANGES_REQUESTED** [review]
doc comment should be changed, and hard coded value should be using a const variable to give more readability

**Comment by clearloop** [comment]
needs guide to configure this pallet into a substrate runtime, mb in https://github.com/ChainSafe/filecoindot/tree/main/substrate-node-example or the README of this pallet

File: filecoindot-nft/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+            // just panic if cannot create class
+            orml_nft::Pallet::<T>::create_class(&owner, vec![], ClassData::new(data))
+                .expect("cannot create default nft class");
+        }
+    }
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        // **************** Relayer Add/Remove *****************
+        /// Adds a new relayer to the relayer set.
```
Comment (clearloop) [inline filecoindot-nft/src/lib.rs:89]: this document is for https://github.com/ChainSafe/filecoindot/blob/a5f96f7b79b41c3c71a3711da136e5413095baa7/filecoindot/src/lib.rs#L246

File: filecoindot-nft/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+                &who,
+                T::DefaultClassId::get(),
+                vec![],
+                TokenData::new(cid, proof),
+            )?;
+            Self::deposit_event(Event::MintedToken(who, T::DefaultClassId::get(), MINT_QUANTITY));
+            Ok(())
+        }
+
+        /// Removes an existing relayer from the set.
```
Comment (clearloop) [inline filecoindot-nft/src/lib.rs:103]: same here

**Review by clearloop — CHANGES_REQUESTED** [review]

**Review by clearloop — APPROVED** [review]
lgtm

please format the code

**Review by ivanceras — APPROVED** [review]
lgtm!

**Pull request merged**